Base/QTCore/qSlicerCoreIOManager.cxx

```diff
@@ -702,6 +709,125 @@ bool qSlicerCoreIOManager::saveNodes(qSlicerIO::IOFileType fileType,
   return true;
 }
 
+//-----------------------------------------------------------------------------
+bool qSlicerCoreIOManager::exportNodes(
+  const QList<QString>& nodeIDs,
+  const QList<QString>& filePaths,
+  const qSlicerIO::IOProperties& parameters,
+  vtkMRMLMessageCollection* userMessages/*=nullptr*/
+)
+{
+  Q_D(qSlicerCoreIOManager);
+  Q_ASSERT(parameters.contains("fileFormat"));
+  Q_ASSERT(nodeIDs.size() == filePaths.size());
+
+  // Create a temporary scene to use for exporting only and to be destroyed when done exporting
+  vtkNew<vtkMRMLScene> temporaryScene;
+  d->currentScene()->CopyDefaultNodesToScene(temporaryScene);
+  d->currentScene()->CopyRegisteredNodesToScene(temporaryScene);
+  d->currentScene()->CopySingletonNodesToScene(temporaryScene);
+
+  // Copy parameters map; we will need to set the nodeID parameter to correspond to nodes in the temporary scene
+  qSlicerIO::IOProperties temporarySceneParameters = parameters;
+
+  bool success = true;
+  for (int i = 0; i<nodeIDs.size(); ++i) {
+
+    // Copy over each node to be exported; shallow copy
+    vtkMRMLNode* node = d->currentScene()->GetNodeByID(nodeIDs[i].toUtf8());
+    if (!node) {
+      if (userMessages)
+        userMessages->AddMessage(vtkCommand::ErrorEvent,
+          (tr("Cannot find the node with ID %1").arg(nodeIDs[i])).toStdString()
+        );
+      success = false;
+      continue;
+    }
+    vtkMRMLNode* temporaryNode = temporaryScene->AddNewNodeByClass(node->GetClassName());
+    if (!temporaryNode) {
+      qCritical() << Q_FUNC_INFO << "Unable to add node to temporary scene";
+      if (userMessages)
+        userMessages->AddMessage(vtkCommand::ErrorEvent,
+          (tr("Error encountered while exporting %1").arg(node->GetName())).toStdString()
+        );
+      success = false;
+      continue;
+    }
+    temporaryNode->CopyContent(node, /*deepCopy=*/false);
```

Comment:
Can you describe an example scenario where transform hardening reuses existing data objects in the node? If possible I'd like to test out a scenario where this happens.

Reply:
A shallow copy does not create an independent copy of an object but some data can be shared. It is up to the node to decide if all data is shared or nothing. If you change anything in the destination object in place (e.g., model point coordinates) and that data happens to be shared with the source object then the source object gets modified, too.

Reply:
Right, I was thinking that this is desirable behavior. But I see now-- you're saying that the transform hardening may modify the underlying data in the temporary node. It didn't in the cases I tested, because I was only messing around with linear transforms. I guess linear transforms on a volume only affect the `vtkImageData` axis orientations and not the underlying data array. I should test this out with transforms that modify the underlying data.

Okay, so make sure to deep copy nodes in the case where underlying data can be modified during export. Got it, thanks.